fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -2085,6 +2098,28 @@ private TransactionSynchronizationRegistry getTrxSynchRegistry() throws FHIRPers
         return allParameters;
     }
 
+    /**
+     * Should we also store values for this {@link SearchParameter} in the special whole-system
+     * param tables (for more efficient whole-system search queries).
+     * @param sp
+     * @return
+     */
+    private boolean isWholeSystem(SearchParameter sp) {
+        boolean result = false;
+
+        // unfortunately we only get a list which means a linear search. Because we're only searching
+        // this once per parameter there's no point in making a Set first, because that would require
+        // a full pass through the list anyway
+        for (ResourceType rt: sp.getBase()) {
+            if ("Resource".equals(rt.getValue())) {
+                result = true;
+                break;
+            }
+        }
```

Comment:
are we planning to replace this with a check for the 4 parameters that will be supported?

if so, we should make sure that any modifier properties (e.g. `token:text`) continue to work as well.

Reply:
Yes...but need to discuss how to do this in the morning.

Reply:
Done